### PR TITLE
Do not import a module you don't need

### PR DESCRIPTION
**Pull request by ryuukk**
I'd personally not call this function at all

### CODE REVIEW

**Comment by CyberShadow** [comment]
Sorry, would you mind explaining in more detail why this an improvement? I'm confused because it deletes one line and adds twenty.

**Comment by CyberShadow** [comment]
I think your argument will be stronger if you include some benchmarks which show the improvement in compilation time brought by this change.

Either way, I really don't think this is a good approach to this general problem. If we take that logic to the extreme, we'd just copy-paste all dependencies into every Phobos module and not have anything import anything.

Such changes may give us small improvements in compilation time, but we will need to keep paying the cost in maintaining duplicate or overcomplicated code.

A perfect solution would be to make the compiler delay parsing of declarations only until it knows it needs them; then, importing big modules just to use one small function would not be a problem at all. In the interim, a better solution may be to move commonly used declarations into tiny modules in an internal namespace, and import those instead.

**Review by CyberShadow — CHANGES_REQUESTED** [review]
I don't think `array = null` here is actually needed at all, you can remove it from the implementations you added.

The problem is that the alternative is a cast, which loses the semantic meaning of the original code. It's a good idea to define correspondingly named functions which do something simple but for a not-entirely-obvious reason (why are we casting away constness? -> because we are moving the only reference (trust me), so it's safe to do so). It's even a good idea to define multiple functions with the exact same body, if the circumstances of using them are different.

Such changes decrease code readability and make the code quality of Phobos worse overall, which is not something we should take lightly.

> We are not yet april 1st right?

I'm trying to hold this conversation in good faith, but, if you're going to be like that, you'll have to convince an organization admin, because I'm out.

File: std/path.d [hunk 1]
```diff
@@ -1773,6 +1772,26 @@ if (isSomeChar!C)
     return () @trusted { return assumeUnique(result.array); } ();
```
Comment (dkorpel) [inline std/path.d:1772]:
```suggestion
    return result.array;
```

**Review by dkorpel — CHANGES_REQUESTED** [review]
The result of `array` can be implicitly converted to immutable, no need for `assumeUnique` here.

**Comment by dkorpel** [comment]
Superseded by https://github.com/dlang/phobos/pull/8726

**Pull request closed without merging**